### PR TITLE
ddl, tools, util: migrate pingcap check to testify.

### LINKED ISSUE #28630: migrate test-infra to testify for `ddl/column_change_test.go`

### LINKED ISSUE #28631: migrate test-infra to testify for `ddl/column_test.go`

### LINKED ISSUE #28638: migrate test-infra to testify for `ddl/ddl_test.go`

### LINKED ISSUE #28639: migrate test-infra to testify for `ddl/ddl_worker_test.go`

### LINKED ISSUE #29104: migrate test-infra to testify for `ddl/index_change_test.go`

### LINKED ISSUE #29108: migrate test-infra to testify for `ddl/reorg_test.go`

### LINKED ISSUE #29109: migrate test-infra to testify for `ddl/restart_test.go`

### LINKED ISSUE #29111: migrate test-infra to testify for `ddl/schema_test.go`

### LINKED ISSUE #29114: migrate test-infra to testify for `ddl/stat_test.go`

### PR DESCRIPTION
**Pull request by GatorMao**
### What problem does this PR solve?

Issue Number: close #29111 ,close #29108 ,close #29104 ,close #29109 ,close #29114 ,close #28639 ,close #28638 ,close #28631,close #28630 

Problem Summary: migrate pingcap check to testify.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by GatorMao** [comment]
Hi @tisonkun, I draft a PR because I encounter few issues during testing and cannot figure out the reason. One of them is the `TestT` test function in ddl package `ddl_test.go` file seems running forever and I don't know the reason for it.
I create suit.Suite field for each test suite and also manually initiate infoSync before each test suite's run method. Not sure if this is the best practice or desired behavior. This is my first contribution to this project and hope could get some help. Thanks !

**Comment by tisonkun** [comment]
@GatorMao thanks for your contribution. The general direction looks good to me. It's a great work!

Please check CI failures and fix it.

**Comment by GatorMao** [comment]
@tisonkun, one of the workflow failed due to following reason. Looks like my change break existing check for test suite in **tools/check/check_testSuite.sh** file. Do you know how to fix it ? Thanks.

`[2022-01-08T01:59:24.226Z] testColumnChangeSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] testStatSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] testSerialStatSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] testIndexChangeSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] testColumnSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] testDDLSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] testDDLSerialSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] testTableSuite in ./ddl is not enabled

[2022-01-08T01:59:24.226Z] make: *** [testSuite] Error 1`

**Comment by GatorMao** [comment]
@tisonkun, all the unit tests pass. Just need to fix the check_testSuite.sh file before submission for final PR.

**Comment by tisonkun** [comment]
@GatorMao rename the suite name should work. It's a legacy check which you can take a look at the script. Don't take it serious :)

**Comment by GatorMao** [comment]
@tisonkun , thanks ! Just rename the test suites and hopefully it works this time.

**Comment by GatorMao** [comment]
@tisonkun, the unit test process seems failed with unrelated reason. My change doesn't touch the tests for br package.

```[2022-01-08T07:55:17.641Z] FAIL: schema_test.go:33: testBackupSchemaSuite.SetUpSuite

[2022-01-08T07:55:17.641Z] 

[2022-01-08T07:55:17.641Z] schema_test.go:37:

[2022-01-08T07:55:17.641Z]     c.Assert(s.mock.Start(), IsNil)

[2022-01-08T07:55:17.641Z] ... value *errors.withStack = listen tcp 0.0.0.0:44642: bind: address already in use ("listen tcp 0.0.0.0:44642: bind: address already in use")

[2022-01-08T07:55:17.641Z] 

[2022-01-08T07:55:17.641Z] OOPS: 7 passed, 1 FAILED, 3 MISSED

**Comment by tisonkun** [comment]
@hawkingrei could you give this PR a review to see whether unit_test's failure is related?

**Comment by zimulala** [comment]
> Details



> @hawkingrei could you give this PR a review to see whether unit_test's failure is related?


This unit_test failure has nothing to do with PR, but a test of BR failed
```
[2022-01-09T21:13:21.958Z]     mock_cluster_test.go:21: 
[2022-01-09T21:13:21.958Z]         	Error Trace:	
[2022-01-09T21:13:21.958Z]         	Error:      	Received unexpected error:
[2022-01-09T21:13:21.958Z]         	            	listen tcp 0.0.0.0:44316: bind: address already in use
[2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/errors.AddStack
[2022-01-09T21:13:21.958Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
[2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/errors.Trace
[2022-01-09T21:13:21.958Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15
[2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/tidb/server.NewServer
[2022-01-09T21:13:21.958Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/server/server.go:245
[2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/tidb/br/pkg/mock.(*Cluster).Start
[2022-01-09T21:13:21.958Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/br/pkg/mock/mock_cluster.go:106
[2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/tidb/br/pkg/mock_test.TestSmoke
[2022-01-09T21:13:21.958Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/br/pkg/mock/mock_cluster_test.go:21
[2022-01-09T21:13:21.958Z]         	            	testing.tRunner
[2022-01-09T21:13:21.958Z]         	            		/usr/local/go/src/testing/testing.go:1193
[2022-01-09T21:13:21.958Z]         	            	runtime.goexit
[2022-01-09T21:13:21.958Z]         	            		/usr/local/go/src/runtime/asm_amd64.s:1371
[2022-01-09T21:13:21.958Z]         	Test:       	TestSmoke
```

**Comment by hawkingrei** [comment]
> > Details
> 
> > @hawkingrei could you give this PR a review to see whether unit_test's failure is related?
> 
> This unit_test failure has nothing to do with this PR, but a test of BR failed
> 
> ```
> [2022-01-09T21:13:21.958Z]     mock_cluster_test.go:21: 
> [2022-01-09T21:13:21.958Z]         	Error Trace:	
> [2022-01-09T21:13:21.958Z]         	Error:      	Received unexpected error:
> [2022-01-09T21:13:21.958Z]         	            	listen tcp 0.0.0.0:44316: bind: address already in use
> [2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/errors.AddStack
> [2022-01-09T21:13:21.958Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
> [2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/errors.Trace
> [2022-01-09T21:13:21.958Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15
> [2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/tidb/server.NewServer
> [2022-01-09T21:13:21.958Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/server/server.go:245
> [2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/tidb/br/pkg/mock.(*Cluster).Start
> [2022-01-09T21:13:21.958Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/br/pkg/mock/mock_cluster.go:106
> [2022-01-09T21:13:21.958Z]         	            	github.com/pingcap/tidb/br/pkg/mock_test.TestSmoke
> [2022-01-09T21:13:21.958Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/br/pkg/mock/mock_cluster_test.go:21
> [2022-01-09T21:13:21.958Z]         	            	testing.tRunner
> [2022-01-09T21:13:21.958Z]         	            		/usr/local/go/src/testing/testing.go:1193
> [2022-01-09T21:13:21.958Z]         	            	runtime.goexit
> [2022-01-09T21:13:21.958Z]         	            		/usr/local/go/src/runtime/asm_amd64.s:1371
> [2022-01-09T21:13:21.958Z]         	Test:       	TestSmoke
> ```



> @hawkingrei could you give this PR a review to see whether unit_test's failure is related?

I find ``` address already in use``` often happen. I can spend some time to find out the cause for it.

**Comment by Mini256** [comment]
@tisonkun, the better practice is to separate multiple linked issues by commas.

```
Issue Number: close #29111, close #29108, close #29104, close #29109, close #29114, close #28639, close #28638, close #28631, close #28630
```

 Ref: [referring-to-an-issue](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/contribute-code.html#referring-to-an-issue)

**Comment by GatorMao** [comment]
Hi @tangenta, would you mind review the ddl_worker_test.go file ? I merged with your change but unit test failed. I don't think my change cause any behavior change for the unit test.

**Comment by tangenta** [comment]
@GatorMao I have filed a PR to stablize it, FYI: https://github.com/pingcap/tidb/pull/31591.

**Comment by GatorMao** [comment]
Much thanks ! @tangenta. Once that PR goes through, I will rebase.

File: ddl/fail_test.go [hunk 1]
```diff
@@ -62,17 +62,17 @@ func (s *testColumnChangeSuite) TestFailBeforeDecodeArgs(c *C) {
 			stateCnt++
 		} else if job.SchemaState == model.StateWriteReorganization {
 			if first {
-				c.Assert(failpoint.Enable("github.com/pingcap/tidb/ddl/errorBeforeDecodeArgs", `return(true)`), IsNil)
+				require.Nil(s.T(), failpoint.Enable("github.com/pingcap/tidb/ddl/errorBeforeDecodeArgs", `return(true)`))
```
[the remark this answers is not in the document]
Comment (GatorMao) [reply ddl/fail_test.go]: Done.

File: ddl/fail_test.go [hunk 2]
```diff
@@ -62,17 +62,17 @@ func (s *testColumnChangeSuite) TestFailBeforeDecodeArgs(c *C) {
...
 				first = false
 			} else {
-				c.Assert(failpoint.Disable("github.com/pingcap/tidb/ddl/errorBeforeDecodeArgs"), IsNil)
+				require.Nil(s.T(), failpoint.Disable("github.com/pingcap/tidb/ddl/errorBeforeDecodeArgs"))
```
[the remark this answers is not in the document]
Comment (GatorMao) [reply ddl/fail_test.go]: Done.

File: ddl/fail_test.go [hunk 3]
```diff
@@ -62,17 +62,17 @@ func (s *testColumnChangeSuite) TestFailBeforeDecodeArgs(c *C) {
...
 		}
 	}
 	d.SetHook(tc)
 	defaultValue := int64(3)
-	job := testCreateColumn(c, ctx, d, s.dbInfo, tblInfo, "c3", &ast.ColumnPosition{Tp: ast.ColumnPositionNone}, defaultValue)
+	job := testCreateColumn(s.T(), ctx, d, s.dbInfo, tblInfo, "c3", &ast.ColumnPosition{Tp: ast.ColumnPositionNone}, defaultValue)
 	// Make sure the schema state only appears once.
-	c.Assert(stateCnt, Equals, 1)
-	testCheckJobDone(c, d, job, true)
+	require.Equal(s.T(), stateCnt, 1)
```
[the remark this answers is not in the document]
Comment (GatorMao) [reply ddl/fail_test.go]: Done.

**Comment by GatorMao** [comment]
@tisonkun , the unit test failure does not come from ddl package. Could you check out and approve my change ? Thanks.

**Comment by GatorMao** [comment]
@Mini256 @hawkingrei @tisonkun , I think the pipeline is in good state right now. Please review and approve this change at your convenience. Thanks.

**Comment by tisonkun** [comment]
@GatorMao thanks for your follow-up. I'm reviewing this PR now.

**Review by tisonkun — APPROVED** [review]
LGTM.

@hawkingrei @xhebox @tangenta PTAL. Since we run all tests in serial, I think using `suite` is acceptable.

**Review by xhebox — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Comment by tisonkun** [comment]
This closes #29102.

**Pull request merged**